### PR TITLE
XYZ fixes

### PR DESCRIPTION
**Pull request by kasper93**

### CODE REVIEW

File: video/csputils.c [hunk 1]
```diff
@@ -769,21 +747,12 @@ void mp_get_csp_matrix(struct mp_csp_params *params, struct mp_cmat *m)
         *m = (struct mp_cmat){{{0, 0, 1}, {1, 0, 0}, {0, 1, 0}}};
         break;
     }
+    case MP_CSP_XYZ:
```
Comment (haasn) [inline video/csputils.c:750]: This change will break XYZ playback in "dumb" renderers like `vo_x11`.

Comment (kasper93) [reply video/csputils.c]: Update this PR, mostly inline with discussion in libplacebo PR. This one mostly fixes software conversion and vo_gpu while at it.

File: filters/f_swscale.c [hunk 2]
```diff
@@ -106,12 +106,6 @@ static void process(struct mp_filter *f)
 
     mp_image_copy_attributes(dst, src);
 
-    // If we convert from RGB to YUV, guess a default.
-    if (mp_imgfmt_get_forced_csp(src->imgfmt) == MP_CSP_RGB &&
-        mp_imgfmt_get_forced_csp(dst->imgfmt) == MP_CSP_AUTO)
-    {
-        dst->params.color.levels = MP_CSP_LEVELS_AUTO;
```
Comment (haasn) [inline filters/f_swscale.c:113]: Why is this change needed? Can you add at least a commit message explaining the justification?

File: video/zimg.c [hunk 3]
```diff
@@ -155,6 +156,9 @@ static zimg_transfer_characteristics_e mp_to_z_trc(enum mp_csp_trc trc)
     case MP_CSP_TRC_GAMMA28:    return ZIMG_TRANSFER_BT470_BG;
     case MP_CSP_TRC_PQ:         return ZIMG_TRANSFER_ST2084;
     case MP_CSP_TRC_HLG:        return ZIMG_TRANSFER_ARIB_B67;
+#if HAVE_ZIMG_ST428
```
Comment (haasn) [inline video/zimg.c:159]: Maybe
```c
#if ZIMG_API_VERSION >= ZIMG_MAKE_API_VERSION(3, 1)
```

File: meson.build [hunk 4]
```diff
@@ -754,6 +754,7 @@ if features['zimg']
         sources += files('test/repack.c',
                          'test/scale_zimg.c')
     endif
+    features += {'zimg-st428': zimg.version().version_compare('>= 3.0.5')}
```
Comment (haasn) [inline meson.build:757]: I don't know how zimg API change policy works but based on the public header only defining the major/minor API version, I think it would have to be a bump to `3.1.0` in order to expose the new enum member?

(In which case, we also don't need to do this check manually)

**Review by haasn — APPROVED** [review]

File: video/zimg.c (hunk 3 above)
Comment (kasper93) [reply video/zimg.c:159]: I wish...

File: meson.build (hunk 4 above)
Comment (kasper93) [reply meson.build:757]: https://github.com/sekrit-twc/zimg/issues/192#issuecomment-1403034902
> Thanks for testing. The API version doesn't get bumped when colorspaces are added. However, the library version will get bumped in the next release, either 3.0.5 or 3.1, and you can check against that.

@haasn: I agree with you, but see above. Current API version is 2.4, current lib version is 3.0.4. API version will not change, so I check for 3.0.5.

File: video/zimg.c (hunk 3 above)
Comment (haasn) [reply video/zimg.c:159]: Sad, but okay

File: filters/f_swscale.c (hunk 2 above)
Comment (kasper93) [reply filters/f_swscale.c:113]: I will update it after work, I think I should remove it, because it is not related to the PR. Although I don't see a reason to not output full range.

Comment (kasper93) [reply filters/f_swscale.c:113]: updated, I left this change in separate commit. I really don't like resetting color params like that, without clear explanation why.

**Comment by haasn** [comment]
LGTM
